### DISCUSSION TITLE
Welcome to OSPP 2024 Discussions!

### DISCUSSION THREAD

**Discussion by faweizhao26** [kubesphere/community#470] [General]
👋 Welcome!
We’re using Discussions as a place to connect with other members of SOPP 2024. We hope that you:

Ask questions you’re wondering about the project of OSPP 2024.

- Show your progress of your project.
- Share ideas.
- Engage with other members.

So you can comment below with your questions or progress or ideas.

**Comment by TriangleWuYQ** [comment]
# Weekly Meeting Report1
- Project: OpenELB
- Reporter: Yuqi Wu
- Progress:
	- Read the OpenELB source code.
	- Complete the testing of IPAM for IPv6 allocation.
	- Partially complete the Neighbor Discovery Protocol.
- Plan:
 	- Complete the NDP (Neighbor Discovery Protocol) and write some corresponding test files. 

# Weekly Meeting Report2
- Project: OpenELB
- Reporter: Yuqi Wu
- Progress:
	- https://github.com/openelb/openelb/pull/442
	- https://github.com/openelb/openelb/pull/443
	- https://github.com/openelb/openelb/pull/445
- Plan:
 	- Complete the advertisement of IPv6 addresses by the OpenELB speaker in BGP mode and VIP mode

# Weekly Meeting Report3
- Project: OpenELB
- Reporter: Yuqi Wu
- Progress:
	- Complete VIP mode advertisement testing for IPv6
- Plan:
 	- Complete BGP mode advertisement testing for IPv6

# Weekly Meeting Report4
- Project: OpenELB
- Reporter: Yuqi Wu
- Progress:
	- Complete BGP mode advertisement testing for IPv6
	- https://github.com/openelb/openelb/pull/448
- Plan:
 	- Write usage documentation for OpenELB in an IPv6 environment.

# Weekly Meeting Report5
- Project: OpenELB
- Reporter: Yuqi Wu
- Progress:
	- Complete boundary testing for IPAM https://github.com/openelb/openelb/pull/449
	- Write user documentation

# Weekly Meeting Report6
- Project: OpenELB
- Reporter: Yuqi Wu
- Progress:
	- Complete Final Report

**Comment by Qi-ming-Zhang** [comment]
## Weekly Meeing Report1

- Project：KubeSphere x LangChain
- Reporter: Qiming Zhang
- Progress:
  - Read KUBESPHERE Extension Component Development Guide and KUBESPHERE API REFERENCE
  - Learning React Framework
  - Detailed Scheduling and Planning for Subsequent Project Phases
- Plan:
  - Identifying Project Requirements and Functional Specifications
  - Defining System Architecture and Technical Solutions
  - Validating Technical Solutions

**Reply by Qi-ming-Zhang** [reply]
## Weekly Meeing Report2

- Project：KubeSphere x LangChain
- Reporter: Qiming Zhang
- Progress:
  - Defining System Architecture and Technical Solutions
  - Validating Technical Solutions
    - Use v0 by Vercel for interface design
    - Retrieve Pod data through the K8s interface
    - Perform data analysis using large language models (such as Qwen2)
    -  Test development process according to the KubeSphere plugin development guide
- Plan:
  - using v0 by Vercel to implement the interface. 
  - Use the React framework to achieve user interaction logic. 
  - Use LangChain.js to display and dynamically update data charts.

**Reply by Qi-ming-Zhang** [reply]
## Weekly Meeing Report3

- Project：KubeSphere x LangChain
- Reporter: Qiming Zhang
- Progress:
  - Complete the validation of the plugin development and deployment process.
  - using v0 by Vercel to implement the interface.
  - Use the React framework to achieve user interaction logic.
- Stage results：
  - PR：Basic static page of pod-analyzer(https://github.com/kubesphere-extensions/ks-extensions-contrib/pull/1)
- Plan:
  - Retrieve Pod data through the K8s interface.
  - Use LangChain.js to display and dynamically update data.
  - Perform data analysis using large language models (such as Qwen2).

**Reply by Qi-ming-Zhang** [reply]
## Weekly Meeing Report4

- Project：KubeSphere x LangChain
- Reporter: Qiming Zhang
- Progress:
  - Retrieve Pod data through the K8s interface.
  - Use LangChain.js to display and dynamically update data.
- Plan:
  - Submit a PR for obtaining and displaying pod data during this cycle.
  - Perform data analysis using large language models (such as Qwen2).

**Reply by Qi-ming-Zhang** [reply]
## Weekly Meeing Report5

- Project：KubeSphere x LangChain
- Reporter: Qiming Zhang
- Progress:
  - Obtain and display pod data during this cycle through the KubeSphere API。
- Stage results：
  - PR：Automated pod information acquisition of pod-analyzer(https://github.com/kubesphere-extensions/ks-extensions-contrib/pull/2)
- Plan:
  - Submit a PR for performing data analysis using large language models during this cycle.

**Reply by Qi-ming-Zhang** [reply]
## Weekly Meeing Report6

- Project：KubeSphere x LangChain
- Reporter: Qiming Zhang
- Progress:
  - Perform data analysis using large language models during this cycle
- Stage results：
  - PR：LLM integration of pod-analyzer(https://github.com/kubesphere-extensions/ks-extensions-contrib/pull/3)
- Plan:
  - Submit a new PR to fix the readme document within this week.

**Comment by jiuxia211** [comment]
## Weekly Meeing Report1

- Project: Fluent Operator
- Reporter: Zixuan Li
- Progress:
  - Install the kubernetes environment and install fluent-operator for test
  - Read the Fluent Operator source code.
- Plan:
  - Complete the update of Fluent Operator dependencies
  - Completed the config-reload enhancement of Fluent bit

**Reply by jiuxia211** [reply]
## Weekly Meeing Report2

- Project: Fluent Operator
- Reporter: Zixuan Li
- Progress:
  - Investigated version updates and usage of multiple dependencies
  - [[Updates multiple dependencie of k8s](https://github.com/fluent/fluent-operator/pull/1251)](https://github.com/fluent/fluent-operator/pull/1251)
- Plan:
  - Further understand and learn the construction of the Fluent Operator project and the functions of each module	
  - Continue completed the config-reload enhancement of Fluent bit

**Reply by jiuxia211** [reply]
## Weekly Meeing Report3

- Project: Fluent Operator
- Reporter: Zixuan Li
- Progress:
  - [[Update fluentbit to v3.1.4](https://github.com/fluent/fluent-operator/pull/1282)](https://github.com/fluent/fluent-operator/pull/1282)
  - [[config-reload Enhancements](https://github.com/fluent/fluent-operator/pull/1286)](https://github.com/fluent/fluent-operator/pull/1286)
- Plan:
  - Integrate Log to Metrics Filter Plugins	
  - Integrate WASM Filter Plugins and WASM Input Plugins

**Reply by jiuxia211** [reply]
## Weekly Meeing Report4

- Project: Fluent Operator
- Reporter: Zixuan Li
- Progress:
  - Understand the mounting of Fluent Operator's configmap and secret, and design the code logic of the wasm plugins
- Plan:
  - Continue integrate WASM Filter Plugins and WASM Input Plugins

**Reply by jiuxia211** [reply]
## Weekly Meeing Report5

- Project: Fluent Operator
- Reporter: Zixuan Li
- Progress:
  - [[add wasm filter piugin](https://github.com/fluent/fluent-operator/pull/1325)](https://github.com/fluent/fluent-operator/pull/1325)
  - [[add exec wasi input plugin](https://github.com/fluent/fluent-operator/pull/1326)](https://github.com/fluent/fluent-operator/pull/1326)
- Plan:
  - Upgrade some new dependencies of fluent bit

**Reply by jiuxia211** [reply]
## Weekly Meeing Report6

- Project: Fluent Operator
- Reporter: Zixuan Li
- Progress:
  - [[Add option to disable operator resources in Helm chart](https://github.com/fluent/fluent-operator/pull/1348#pullrequestreview-2312206620)](https://github.com/fluent/fluent-operator/pull/1348)
- Plan:
  - Fix correct import path for v3 version of fluent-operator

**Comment by dbbDylan** [comment]
## Weekly Meeting Report 1
- Project: KubeKey
- Reporter: Yongliang Ding
- Progress:
  - Set up and configure the Go language and Kubernetes development environment.
  - Learn the basics of Go and Kubernetes.
  - Read the KubeKey source code, further understanding the Cluster API Doc.
- Plan:
  - Complete the environment setup and testing for KubeKey.
  - Attempt to build a KKMachine lifecycle manager framework.
  - Attempt to integrate Cluster API into KubeKey and complete some test cases.

**Reply by dbbDylan** [reply]
# Weekly Meeting Report 2
- Project: KubeKey
- Reporter: Yongliang Ding
- Progress:
  - Complete the environment setup and testing for KubeKey.
  - Using `clusterctl` command at master node completing the API test. 
- Plan:
  - Continue to understanding KubeKey and Cluster API Doc.
  - Continue to integrate `clusterctl` command into KubeKey.

**Reply by dbbDylan** [reply]
## Weekly Meeting Report 3
- Project: KubeKey
- Reporter: Yongliang Ding
- Progress:
  - Modify Kubekey templates and profiles, and complete k8s-cluster tests. [https://github.com/kubesphere/kubekey/pull/2346](https://github.com/kubesphere/kubekey/pull/2346).
  - Generate KKCluster and KKMachine CRD&Controller template, preliminary design of the CAPKK implementation process.
- Plan:
  - Continue to modify and optimize kk's playbook templates and profiles.
  - Complete the v1beta1 version CAPKK.

**Reply by dbbDylan** [reply]
## Weekly Meeting Report 4
- Project: KubeKey
- Reporter: Yongliang Ding
- Progress:
  - Complete: [https://github.com/kubesphere/kubekey/pull/2364](https://github.com/kubesphere/kubekey/pull/2364).
- Plan:
  - Continue to modify and optimize kk's playbook templates and profiles.
  - Continue to complete reconcile functions.

**Reply by dbbDylan** [reply]
## Weekly Meeting Report 5
- Project: KubeKey
- Reporter: Yongliang Ding
- Progress:
  - Implement node check and selector capacity, and join it into CRD reconcile.
  - Basically implement cluster create normal reconcile stage, testing.
- Plan:
  - Refactor and integrate existing code, submit relevant PR
  - Complete the rest of the functional development

**Reply by dbbDylan** [reply]
## Weekly Meeting Report 6
- Project: KubeKey
- Reporter: Yongliang Ding
- Progress:
  - https://github.com/kubesphere/kubekey/pull/2406
  - https://github.com/kubesphere/kubekey/pull/2409
  - https://github.com/kubesphere/kubekey/pull/2410
- Plan:
  - Complete Kubernetes certs' distribution function for the cluster, replace certs generated by playbook now.
  - Fix bug for the playbooks.
  - Complete README.md & relevant documents.
